domain-models-runtime/src/main/java/org/folio/rest/RestVerticle.java

```diff
@@ -1339,7 +1339,7 @@ else if(bodyContent.length() > 0) {
                 if (defaultVal != null) {
                   paramArray[order] = Integer.valueOf((String) defaultVal);
                 } else {
-                  paramArray[order] = 0;
+                  paramArray[order] = valueType.equals("int") ? 0 : null;
```

Comment:
I was thinking parameter type support could be more customizable. Something like parameter processors. Have default parameter processors in RMB and support custom parameter processors from modules with support for custom types.

This would make parameter support much more extensible and could help reduce number of arguments in endpoint methods.

I would much rather do that than just extract less than ideal code into a method to just be able to unit test it.

Reply:
Added integer query parameter to books API which covers the case for integer primitive query parameter. Added an example wrapped integer API and tests which covers the wrapped integer cases. This should cover the lines well enough. RestVerticle is in need of unit testing and possibly refactoring, but I feel that is out of scope of this issue.